packages/seacas/libraries/ioss/src/exodus/Ioex_DatabaseIO.C

```diff
@@ -641,74 +641,85 @@ namespace Ioex {
     }
   }
 
-  void DatabaseIO::get_step_times_nl()
+  std::vector<double> DatabaseIO::internal_get_step_times_nl(bool setRegionTimeSteps)
   {
     bool                exists         = false;
     double              last_time      = DBL_MAX;
+    int                 tstepCount     = 0;
     std::vector<double> tsteps(0);
 
+    int& timestepCount = setRegionTimeSteps ? m_timestepCount : tstepCount;
+
     if (dbUsage == Ioss::WRITE_HISTORY) {
       if (myProcessor == 0) {
-        m_timestepCount = ex_inquire_int(get_file_pointer(), EX_INQ_TIME);
-        if (m_timestepCount <= 0) {
-          return;
+        timestepCount = ex_inquire_int(get_file_pointer(), EX_INQ_TIME);
+        if (timestepCount <= 0) {
+          return tsteps;
         }
 
         // For an exodus file, timesteps are global and are stored in the region.
         // A history file only stores that last time / step
         // Read the timesteps and add them to the region.
         // Since we can't access the Region's stateCount directly, we just add
         // all of the steps and assume the Region is dealing with them directly...
-        tsteps.resize(m_timestepCount);
+        tsteps.resize(timestepCount);
 
         int error = ex_get_all_times(get_file_pointer(), Data(tsteps));
         if (error < 0) {
           Ioex::exodus_error(get_file_pointer(), __LINE__, __func__, __FILE__);
         }
 
-        int max_step = properties.get_optional("APPEND_OUTPUT_AFTER_STEP", m_timestepCount);
-        max_step     = std::min(max_step, m_timestepCount);
+        int max_step = properties.get_optional("APPEND_OUTPUT_AFTER_STEP", timestepCount);
+        max_step     = std::min(max_step, timestepCount);
 
         double max_time =
             properties.get_optional("APPEND_OUTPUT_AFTER_TIME", std::numeric_limits<double>::max());
 
         Ioss::Region *this_region = get_region();
+        int numSteps = 0;
         for (int i = 0; i < max_step; i++) {
           if (tsteps[i] <= max_time) {
-            this_region->add_state_nl(tsteps[i] * timeScaleFactor);
+            if(setRegionTimeSteps) {
+              this_region->add_state_nl(tsteps[i] * timeScaleFactor);
+            }
+
+            tsteps[i] *= timeScaleFactor;
```

Comment:
@gsjaardema It is unclear if to return the original time or the scaled time ...

Reply:
I think scaled time.